app/js/lib/foxbox/services/base.js

```diff
@@ -290,12 +291,22 @@ export default class BaseService extends EventDispatcher {
    *
    * @param {Object|string} selector Selector that channel should match to.
    * @return {Object}
+   * @throws Will throw if there is no channel that corresponds to selector and
+   * supports fetch operation.
    * @private
    */
   [p.getFetchChannel](selector) {
-    return this[p.getChannels](selector).find(
+    const channel = this[p.getChannels](selector).find(
       (channel) => channel.supports_fetch
     );
+
+    if (!channel) {
```

Comment:
I found these error messages useful when integrating new watchers/getter/setters for services.